implementation/src/main/java/io/smallrye/metrics/app/SimpleTimerImpl.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ * *******************************************************************************
+ * Copyright 2010-2013 Coda Hale and Yammer, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.smallrye.metrics.app;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.eclipse.microprofile.metrics.SimpleTimer;
+
+public class SimpleTimerImpl implements SimpleTimer {
+
+    private final Clock clock;
+
+    // total number of captured measurements
+    private final AtomicLong count;
```

Comment:
Would `LongAdder` be more suitable for this purpose? (Perhaps in other classes too, not sure, didn't look.)

Reply:
Hmm. Interesting point. From what I read, `LongAdder` is faster under high contention, but at the expense of higher memory consumption (but that occurs only when there is high contention). Under low contention, `AtomicLong` seems a bit faster. `LongAdder` is also suitable for cases when there's a lot of updates, but not too many reads. This could potentially be our case, because metrics will typically be scraped once per 15 seconds or so. Ok, I'll switch to `LongAdder`. `HistogramImpl` is using it too.